scripts/visualize_learning_rate.py

```diff
@@ -0,0 +1,111 @@
+import sys
+from pathlib import Path
+from matplotlib.pyplot import plot
+
+from torch import optim
+sys.path.append(str(Path(__file__).parents[1].joinpath('src', 'development')))
+
+import argparse
+import warnings
+import torch
+import numpy as np
+import matplotlib.pyplot as plt
+
+warnings.filterwarnings('ignore')
+from vortex.development.utils.parser import load_config
+from vortex.development.core.engine.trainer import lr_scheduler
+warnings.resetwarnings()
+
+
+def calculate_lr(lr_config, epochs=100, optimizer_config=None):
+    ## optimizer first
+    if optimizer_config is None:
+        optimizer_config = { ## just put a dummy optimizer
+            'module': 'SGD',
+            'args': {
+                'lr': 0.01,
+                'momentum': 0.95,
+                'weight_decay': 0.0001,
+            }
+        }
+
+    if 'method' in optimizer_config and not 'module' in optimizer_config:
+        optimizer_config.update({'module': optimizer_config['method']})
+    opt_method, kwargs = optimizer_config['module'], optimizer_config['args']
+    if opt_method.startswith('optim.'):
+        opt_method = opt_method.replace('optim.','')
+    assert hasattr(optim, opt_method), "unsupported optimizer {}".format(opt_method)
+    kwargs.update({'params' : torch.nn.Linear(2, 1).parameters()})  ## dummy model
+    optimizer = getattr(optim, opt_method)(**kwargs)
+
+    ## lr scheduler
+    if 'method' in lr_config and not 'module' in lr_config:
+        lr_config.update({'module': lr_config['method']})
+    sch_method, kwargs = lr_config['module'], lr_config['args']
+    assert hasattr(lr_scheduler, sch_method), "unsupported lr_scheduler {}".format(sch_method)
+    kwargs.update({'optimizer': optimizer})
+    scheduler = getattr(lr_scheduler, sch_method)(**kwargs)
+
+    print("Visualizing {} lr scheduler for {} epoch".format(sch_method, epochs))
+    lr_data = []
+    for ep in range(epochs):
+        scheduler.step(ep)
+        lr_data.append(scheduler.get_lr()[0])
+    return np.array(lr_data)
+
+
+def visualize(learning_rates):
+    pass
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="update old model checkpoint to the new format")
+    parser.add_argument('CFG', type=str, nargs='?', help="configuration file path (.yml)")
+    parser.add_argument('-c', '--config', required=False, type=str,
+        help="configuration file path (choose either one of this or positional argument)")
+    parser.add_argument('--epochs', required=False, type=int, 
```

Comment:
I think this args should be parsed from experiment config too

Reply:
it would be easier if we could override from the config, if we want to experiment with it.  
also it is not required really, and will use the data from the config if not provided

Reply:
hoo i see, if not provided by default it fetch from experiment config right? cool

Reply:
yepp, that's correct